slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -382,20 +411,15 @@ std::string SLAMServiceImpl::GetNextDataFile() {
     return GetNextDataFileOnline();
 }
 
-void SLAMServiceImpl::CreateMap() {
+void SLAMServiceImpl::RunSLAM() {
     cartographer::mapping::TrajectoryBuilderInterface *trajectory_builder;
     int trajectory_id;
     {
         std::lock_guard<std::mutex> lk(map_builder_mutex);
-        // Build TrajectoryBuilder
-        trajectory_id = map_builder.map_builder_->AddTrajectoryBuilder(
-            {kRangeSensorId}, map_builder.trajectory_builder_options_,
-            map_builder.GetLocalSlamResultCallback());
-
-        LOG(INFO) << "Trajectory ID: " << trajectory_id;
-
-        trajectory_builder =
-            map_builder.map_builder_->GetTrajectoryBuilder(trajectory_id);
+        // Set TrajectoryBuilder
+        trajectory_id = map_builder.SetTrajectoryBuilder(&trajectory_builder,
+                                                         {kRangeSensorId});
+        LOG(INFO) << "Using trajectory ID: " << trajectory_id;
```

Comment:
[opt]  `Using new trajectory ID:` (if comment from early is taken to add from apriori map to log of other trajectory ids)